tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -1,32 +1,493 @@
+                options.SignUpSignInPolicyId = TestConstants.B2CSignUpSignInUserFlow;
+                options.Domain = TestConstants.Domain;
+            };
+
+            var services = new ServiceCollection();
+            services.AddDataProtection();
+
+            if (useServiceCollectionExtension)
+                services.AddSignIn(_configureOidcOptions, _configureMsOptions, _oidcScheme, _cookieScheme, false);
+            else
+                new AuthenticationBuilder(services)
+                    .AddSignIn(_configureOidcOptions, _configureMsOptions, _oidcScheme, _cookieScheme, false);
+
+            await AddSignIn_TestB2cSpecificSetup(services).ConfigureAwait(false);
+        }
+
+        private async Task AddSignIn_TestB2cSpecificSetup(IServiceCollection services)
+        {
+            var provider = services.BuildServiceProvider();
+
+            var oidcOptions = provider.GetRequiredService<IOptionsFactory<OpenIdConnectOptions>>().Create(_oidcScheme);
+
+            // Assert B2C name claim type
+            Assert.Equal(ClaimConstants.Name, oidcOptions.TokenValidationParameters.NameClaimType);
+
+            var httpContext = HttpContextUtilities.CreateHttpContext();
+            var authScheme = new AuthenticationScheme(OpenIdConnectDefaults.AuthenticationScheme, OpenIdConnectDefaults.AuthenticationScheme, typeof(OpenIdConnectHandler));
+            var authProperties = new AuthenticationProperties();
+            authProperties.Items[OidcConstants.PolicyKey] = TestConstants.B2CResetPasswordUserFlow;
+            var redirectContext = new RedirectContext(httpContext, authScheme, oidcOptions, authProperties)
+            {
+                ProtocolMessage = new OpenIdConnectMessage() { IssuerAddress = $"IssuerAddress/{TestConstants.B2CSignUpSignInUserFlow}/" },
+            };
+
+            await oidcOptions.Events.RedirectToIdentityProvider(redirectContext).ConfigureAwait(false);
+
+            // Assert issuer is updated to non-default user flow
+            Assert.Contains(TestConstants.B2CResetPasswordUserFlow, redirectContext.ProtocolMessage.IssuerAddress);
+        }
+
+        private IConfigurationSection GetConfigSection(string configSectionName, bool includeB2cConfig = false)
+        {
+            var configAsDictionary = new Dictionary<string, string>()
             {
-                Authority = initialAuthority
+                { configSectionName, null },
+                { $"{configSectionName}:Instance", TestConstants.AadInstance },
+                { $"{configSectionName}:TenantId", TestConstants.TenantIdAsGuid },
+                { $"{configSectionName}:ClientId", TestConstants.TenantIdAsGuid },
+                { $"{configSectionName}:Domain", TestConstants.Domain },
             };
 
-            options.Authority = AuthorityHelpers.EnsureAuthorityIsV2(options.Authority);
-            Assert.Equal(expectedAuthority, options.Authority);
+            if (includeB2cConfig)
+            {
```

Comment:
needs to include b2c specific instance and domain

Reply:
Specific value doesn't affect the outcome of the test but I've updated it for readability.